### ISSUE TITLE
Add determines whether the subscriber parameter is interface instance

### ISSUE THREAD

**Issue by yijianmin** [dotnetcore/CAP#669]
Use IEnumerable to publish to RabbitMQ, and then the consumer uses IEnumerable to receive and report the above exception.

The release is successful, but the consumer does not consume it. The cap.received table is recorded as Failed, and Retries is directly displayed as 3. After more than 4 minutes, consumption starts, and then the cap.received table record is changed to Succeeded.

Finally, replace IEnumerable with List. There is no such problem, no delay and no exception. What is the reason?

```
DotNetCore.CAP.Internal.SubscriberExecutionFailedException: Object must implement IConvertible.
System.InvalidCastException: Object must implement IConvertible.
at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
at DotNetCore.CAP.Internal.SubscribeInvoker.InvokeAsync(ConsumerContext context, CancellationToken cancellationToken)
at DotNetCore.CAP.Internal.SubscribeDispatcher.InvokeConsumerMethodAsync(MediumMessage message, ConsumerExecutorDescriptor descriptor, CancellationToken cancellationToken)
```

**Comment by yang-xiaodong** [comment]
This is a basic knowledge, the data needs to be carried by the object instance not interface

```C#
var obj = JsonConvert.DeserializeObject("[\"2020-01-01\"]", typeof(IEnumerable<DateTime>));
Console.WriteLine(obj.GetType()); // Output : System.Collections.Generic.List`1[System.DateTime]
```

**Comment by yang-xiaodong** [comment]
I think this is an item that can be improved. Do this by passing to check if the instance and interface match

**Comment by yang-xiaodong** [comment]
Fixed in v3.1.1

**Comment by jirisykora83** [comment]
Facing exact same issue with last version. Frist 3 times in row (almost instantly) got `DotNetCore.CAP.Internal.SubscriberExecutionFailedException: Object must implement IConvertible` and then after some time (by default 4min) message was successfully process (without changing anything even without restarting etc..).

```
DotNetCore.CAP.Internal.SubscriberExecutionFailedException: Object must implement IConvertible.
 ---> System.InvalidCastException: Object must implement IConvertible.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider) in System.Private.CoreLib.dll:token 0x6000b96+0x0
   at DotNetCore.CAP.Internal.SubscribeInvoker.InvokeAsync(ConsumerContext context, CancellationToken cancellationToken) in DotNetCore.CAP.dll:token 0x60001d1+0x25f
   at DotNetCore.CAP.Internal.SubscribeDispatcher.InvokeConsumerMethodAsync(MediumMessage message, ConsumerExecutorDescriptor descriptor, CancellationToken cancellationToken) in DotNetCore.CAP.dll:token 0x60001ca+0xb9
   --- End of inner exception stack trace ---
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider) in System.Private.CoreLib.dll:token 0x6000b96+0x0
   at DotNetCore.CAP.Internal.SubscribeInvoker.InvokeAsync(ConsumerContext context, CancellationToken cancellationToken) in DotNetCore.CAP.dll:token 0x60001d1+0x25f
   at DotNetCore.CAP.Internal.SubscribeDispatcher.InvokeConsumerMethodAsync(MediumMessage message, ConsumerExecutorDescriptor descriptor, CancellationToken cancellationToken) in DotNetCore.CAP.dll:token 0x60001ca+0xb9
```

---

After some investigation i found that issue is when in my case I have `ConsumerServiceSelector` which register my handler automatically and method which handle Event have second parameter with `CancellationToken`

```
public class TestHandler : BaseAsyncHandler<TestEvent>
{
  public override async Task Handle(FullCacheInvalidateEvent notification,
	  CancellationToken cancellationToken)
  {
	  
  }
  }
```
  
  Should be to ekvivalent to something like this (i do not try it)

```
[CapSubscribe("tomethind")
  public override async Task Handle(FullCacheInvalidateEvent notification,
	  CancellationToken cancellationToken)
  {
	
  }
```
